### PR TITLE
Add args() getter/setter generics for task-graph nodes

### PR DESCRIPTION
**Pull request by johnkerl**
Rather than `d$set_args(list(a,b,c))`, we can now do the more R-idiomatic `args(d) <- list(a,b,c)`.

### CODE REVIEW

File: R/manual_layer_node.R [hunk 1]
```diff
@@ -367,7 +367,35 @@ setMethod("compute_sequentially", signature(object = "Node"), function(object) {
   object$compute_sequentially()
 })
 
-# ----------------------------------------------------------------
+##' Get arguments for a delayed function, as a list.
+##'
+##' @family {manual-layer functions}
+##' @export
+args <- function(object, ...) 0
```
Comment (eddelbuettel) [inline R/manual_layer_node.R:374]: That 0 looks like a typo here.  Can you make it `NULL` or just (maybe more idiomatic) `{}`  ?

`args()` is an existing base R function so maybe the naming here is too aggressive as it shadows a base function? But it looks like your use of a generic added here prevents harm.  If that does in fact work, great!   (May be prudent to have a unit test ir or two 'calibrated' on base R's `args()` in the package to ensure it still works after package load? Or maybe that is overkill.

File: R/manual_layer_node.R [hunk 2]
```diff
@@ -367,7 +367,35 @@ setMethod("compute_sequentially", signature(object = "Node"), function(object) {
...
+##'
+##' @family {manual-layer functions}
+##' @export
+"args<-" <- function(object, ...) 0
+# Note: the "..." signature is forced on us.
+setMethod("args<-", signature(object = "Node"), function(object, ...) {
+  stopifnot(...length() == 1)
+  args <- list(...)
+  stopifnot(is.list(args))
+  stopifnot(is.list(args$value)) # the <- logic gives us a list with single name "value"
```
Comment (eddelbuettel) [inline R/manual_layer_node.R:394]: `stopifnot()` can use multiple expressions at once, and also embed the emitted warning -- a scheme I like a lot. I suggest

```r
stopifnot(`argument must be a list object` = is.list(args),
          `value component must be a list` = is.list(args$value))
```

and I would suggest to also test if component `value` exists among the name (`"value" %in% names(args)` may do, or maybe `match`).

**Review by eddelbuettel — COMMENTED** [review]
See inline

File: R/manual_layer_node.R (hunk 1 above)
Comment (eddelbuettel) [reply R/manual_layer_node.R:374]: It is indeed a problem as I suspected but I once again fell into the rabbit hole of looking at this before other commit came in.   As it stands right now, R is pretty clear about this:

```r
> library(tiledbcloud)

Attaching package: ‘tiledbcloud’

The following object is masked from ‘package:base’:

    args

> 
```

**Review by eddelbuettel — CHANGES_REQUESTED** [review]
Per follow-up, I would suggest to change the naming of `args`.

**Comment by johnkerl** [comment]
Thanks @eddelbuettel -- https://github.com/TileDB-Inc/TileDB-Cloud-R/pull/67/commits/949c1d3f1aa07e29adc817bd475303a540ab05e2

**Comment by johnkerl** [comment]
Rebased atop https://github.com/TileDB-Inc/TileDB-Cloud-R/pull/63 which is now merged

**Review by eddelbuettel — DISMISSED** [review]

File: vignettes/TaskGraphs.Rmd [hunk 3]
```diff
@@ -14,14 +14,14 @@ to be executed later, with a few more items:
 
 * Optionally a friendly display name in `name`
 * Optionally `local=TRUE` to execute the function on the current host -- the default is to execute on TileDB Cloud.
-* If the function takes arguments, they can be specified by `args=...` in `delayed`, or, via a separate `set_args`.
+* If the function takes arguments, they can be specified by `args=...` in `delayed`, or, via a separate call to `args`.
```
[the remark this answers is not in the document]
Comment (johnkerl) [reply vignettes/TaskGraphs.Rmd]: thanks @eddelbuettel !

**Review by eddelbuettel — APPROVED** [review]
Looks good.  The only thing I scratch my head about is whether there are S4 + R6 mixup issues  but nobody says we can't, and the higher-level setter is nice.

**Pull request merged**